plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectionCreation.java

```diff
@@ -0,0 +1,110 @@
+
+import java.util.List;
+import java.util.Properties;
+
+import static io.trino.plugin.jdbc.H2QueryRunner.createH2QueryRunner;
+import static io.trino.tpch.TpchTable.NATION;
+import static io.trino.tpch.TpchTable.REGION;
+import static java.util.Objects.requireNonNull;
+
+@Test(singleThreaded = true) // this class is stateful, see fields
+public class TestJdbcConnectionCreation
+        extends BaseJdbcConnectionCreationTest
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        // Exercise information_schema.columns with more than one table
+        String connectionUrl = TestingH2JdbcModule.getConnectionUrl();
+        DriverConnectionFactory delegate = new DriverConnectionFactory(new Driver(), connectionUrl, new Properties(), new EmptyCredentialProvider());
+        this.connectionFactory = new ConnectionCountingConnectionFactory(delegate);
+        return createH2QueryRunner(ImmutableList.of(NATION, REGION), ImmutableMap.of("connection-url", connectionUrl), new ReusingConnectionH2Module(connectionFactory));
+    }
+
+    @Test
+    public void dummyTestToMakeSureTestNgRunsTestsSerially() {}
+
+    @Override
+    protected List<TestCase> getTestCases()
+    {
+        return ImmutableList.of(
+                new TestCase("SELECT * FROM nation LIMIT 1", 3),
+                new TestCase("SELECT * FROM nation ORDER BY nationkey LIMIT 1", 3),
+                new TestCase("SELECT * FROM nation WHERE nationkey = 1", 3),
+                new TestCase("SELECT avg(nationkey) FROM nation", 3),
+                new TestCase("SELECT * FROM nation, region", 6),
+                new TestCase("SELECT * FROM nation n, region r WHERE n.regionkey = r.regionkey", 6),
+                new TestCase("SELECT * FROM nation JOIN region USING(regionkey)", 6),
+                new TestCase("SELECT * FROM information_schema.schemata", 1),
+                new TestCase("SELECT * FROM information_schema.tables", 1),
+                new TestCase("SELECT * FROM information_schema.columns", 5),
+                new TestCase("SELECT * FROM nation", 3),
+                new TestCase("CREATE TABLE copy_of_nation AS SELECT * FROM nation", 13),
+                new TestCase("INSERT INTO copy_of_nation SELECT * FROM nation", 14),
+                new TestCase("DELETE FROM copy_of_nation WHERE nationkey = 3", 3),
+                new TestCase("UPDATE copy_of_nation SET name = 'POLAND' WHERE nationkey = 1", 2, "This connector does not support updates"),
+                new TestCase("MERGE INTO copy_of_nation n USING region r ON r.regionkey= n.regionkey WHEN MATCHED THEN DELETE", 2, "This connector does not support merges"),
+                new TestCase("DROP TABLE copy_of_nation", 3),
+                new TestCase("SHOW SCHEMAS", 1),
+                new TestCase("SHOW TABLES", 2),
+                new TestCase("SHOW STATS FOR nation", 2));
+    }
+
+    private static class ReusingConnectionH2Module
```

Comment:
I don't see any reuse here.
Where should I be looking?

Reply:
I extracted this commit from reuse connection.